src/rsa/padding.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015-2016 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use untrusted;
+use digest;
+
+#[cfg(feature = "rsa_signing")]
+use rand;
+
+#[cfg(feature = "rsa_signing")]
+pub trait Encoding {
```

Comment:
I'm not sure why you choose to rename `Padding` to `Encoding` here, but I think I don't like it that much. Similarly, I don't think `encode()` is the right name for the method. It seems to me that `Padding` is probably still the best name for the trait, and the method should either be `pad()` or maybe `wrap()` (the latter mostly because for the inverse, `unwrap()` maybe is nicer than `unpad()`).